robottelo/enums.py

```diff
@@ -0,0 +1,35 @@
+"""Module containing enumeration classes used throughout Robottelo.
+
+This module provides standardized enumerations for various types, statuses,
+and configurations used in Robottelo tests and utilities.
+"""
+
+from enum import StrEnum
+
+
+class NetworkType(StrEnum):
```

Comment:
Writing it out as an idea that I mentioned elsewhere: implement `has_ipv4` and `has_ipv6`. Very often the pattern is that if no IPv4 is present then some HTTP proxy must be used. So

```python
@property
def has_ipv4(self):
    return self in (IPV4, DUALSTACK)

@property
def has_ipv6(self):
    return self in (IPV6, DUALSTACK)
```

Then in the code it's best to write:
```python
if not network_type.has_ipv4:
    enable_http_proxy()
```

Or in the provisioning case:
```python
if network_type.has_ipv4:
    subnet = Subnet()
else:
    subnet = None

if network_type.has_ipv6:
    subnet6 = Subnet()
else:
    subnet6 = None
```

IMHO this allows for better dual stack implementations.